src/zeta/aca_oam_port_manager.cpp

```diff
@@ -83,101 +83,119 @@ void Aca_Oam_Port_Manager::_clear_all_data()
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::clear_all_data ---> Entering\n");
 
   // -----critical section starts-----
-  _oam_ports_table_mutex.lock();
-  // All the elements in the unordered_map container are dropped:
+  _oam_ports_cache_mutex.lock();
+  // All the elements in the unordered_set container are dropped:
   // their destructors are called, and they are removed from the container,
-  // leaving _oam_ports_table with a size of 0.
-  _oam_ports_table.clear();
-  _oam_ports_table_mutex.unlock();
+  // leaving _oam_ports_cache with a size of 0.
+  _oam_ports_cache.clear();
+  _oam_ports_cache_mutex.unlock();
   // -----critical section ends-----
 
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::clear_all_data <--- Exiting\n");
 }
 
 // unsafe function, needs to be called inside oam_ports_table_mutex lock
 // this function assumes there is no existing entry for port_number
-void Aca_Oam_Port_Manager::create_entry_unsafe(uint32_t port_number)
+void Aca_Oam_Port_Manager::create_entry_unsafe(uint port_number)
 {
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe ---> Entering\n");
 
-  unordered_set<uint32_t> tunnel_ids_table;
-  _oam_ports_table.emplace(port_number, tunnel_ids_table);
+  _oam_ports_cache.emplace(port_number);
 
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe <--- Exiting\n");
 }
 
 // update oam_ports_table and add the OAM punt rule also if this is the first port in the VPC
-void Aca_Oam_Port_Manager::add_vpc(uint32_t port_number, uint32_t tunnel_id)
+void Aca_Oam_Port_Manager::add_oam_port_rule(uint port_number)
 {
-  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_vpc ---> Entering\n");
+  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_oam_port_rule ---> Entering\n");
   // -----critical section starts-----
-  _oam_ports_table_mutex.lock();
-  if (_oam_ports_table.find(port_number) == _oam_ports_table.end()) {
+  _oam_ports_cache_mutex.lock();
+  if (_oam_ports_cache.find(port_number) == _oam_ports_cache.end()) {
```

Comment:
the _oam_ports_cache is only use for ACA_OVS_Control_parse_packet function for the quick check to see if it is an OAM port UDP packet. Additional operation like _create_oam_ofp or _delete_oam_ofp should be done updating the _vpcs_table with oam port information.